src/main/java/org/opentripplanner/middleware/controllers/api/LogController.java

```diff
@@ -1,29 +1,76 @@
 package org.opentripplanner.middleware.controllers.api;
 
 import com.amazonaws.services.apigateway.model.GetUsageResult;
+import com.beerboy.ss.ApiEndpoint;
+import com.beerboy.ss.SparkSwagger;
+import com.beerboy.ss.rest.Endpoint;
 import org.eclipse.jetty.http.HttpStatus;
 import org.opentripplanner.middleware.utils.ApiGatewayUtils;
 import org.opentripplanner.middleware.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import spark.Request;
 import spark.Response;
-import spark.Service;
 
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.List;
 
+import static com.beerboy.ss.descriptor.EndpointDescriptor.endpointPath;
+import static com.beerboy.ss.descriptor.MethodDescriptor.path;
+import static org.opentripplanner.middleware.utils.DateUtils.STD_DATE_PATTERN;
+import static org.opentripplanner.middleware.utils.HttpUtils.MIMETYPES_JSONONLY;
 import static org.opentripplanner.middleware.utils.JsonUtils.logMessageAndHalt;
 
 /**
  * Sets up HTTP endpoints for getting logging and request summary information from AWS Cloudwatch and API Gateway.
  */
-public class LogController {
+public class LogController implements Endpoint {
+    private static final Logger LOG = LoggerFactory.getLogger(LogController.class);
+    private final String ROOT_ROUTE;
+
+    public LogController(String apiPrefix) {
+        this.ROOT_ROUTE = apiPrefix + "secure/logs";
+    }
 
     /**
-     * Register http endpoints with {@link spark.Spark} instance at the provided API prefix.
+     * Register the API endpoint and GET resource to retrieve API usage logs
+     * when spark-swagger calls this function with the target API instance.
      */
-    public static void register(Service spark, String apiPrefix) {
-        spark.get(apiPrefix + "/secure/logs", LogController::getUsageLogs, JsonUtils::toJson);
+    @Override
+    public void bind(final SparkSwagger restApi) {
+        ApiEndpoint apiEndpoint = restApi.endpoint(
+            endpointPath(ROOT_ROUTE).withDescription("Interface for retrieving API logs from AWS."),
+            (q, a) -> LOG.info("Received request for 'logs' Rest API")
+        );
+        apiEndpoint
+            .get(path(ROOT_ROUTE)
+                    .withDescription("Gets a list of all API usage logs.")
+                    .withQueryParam()
+                        .withName("keyId")
+                        .withDescription("If specified, restricts the search to the specified AWS API key ID.").and()
+                    .withQueryParam()
+                        .withName("startDate")
+                        .withPattern(STD_DATE_PATTERN)
```

Comment:
I don't love the name of `STD_DATE_PATTERN`... let's just rename this to `YYYY_MM_DD`.

Reply:
Fixed in https://github.com/ibi-group/otp-middleware/pull/55/commits/e09acb4c987fcb145048f91cff1d1fa13374a7bf